content/usage/advanced/index.md

```diff
@@ -284,94 +538,197 @@ being redirected via BlueOS, but can also be used to display online videos if th
+various [protocol functions](#joystick-protocols), which can send inputs and commands to the vehicle, or
+trigger interface events.
+
+{% note() %}
+The default function mapping is selected based on the connected vehicle type, and Cockpit automatically
+synchronises function mappings to/from the [User](#users), but it is also possible to manually export 
+them as a file, and import them into another User or a different Cockpit instance/device.
+{% end %}
+
+Known joystick types have an interactive diagram for mapping button and axis functions visually:
+{{ easy_image(src="joystick-button-mapping", width=600, center=true) }}
+
+Button presses and axis movements should be mirrored on the diagram, and clicking on a button
+element in the diagram allows remapping its mapped function.
 
-{{ easy_image(src="../getting-started/joystick-config", width=600, center=true) }}
+{% note() %}
+Axes currently must have unique functions, so mapping an axis to an Action that is already in
+use will result in that Action being unmapped from whichever axis was using it previously.
+{% end %}
+
+There is also a table view, which is available for both known and new joystick types:
+{{ easy_image(src="joystick-table", width=600, center=true) }}
+
+Buttons can be remapped to different Actions using the edit pencil at the far right of the corresponding row.
 
+{% note() %}
 Support is built in for simultaneous input from multiple sources, including multiple joysticks, and by
-default each joystick can provide up to 8 axis ranges and 32 buttons.
+default each joystick can provide up to 32 axis ranges and 32 buttons.
+{% end %}
 
 #### Joystick Protocols
 
 When mapping the functionality of a joystick button or axis, there are multiple protocols to choose from:
 
-{{ easy_image(src="joystick-button-mapping", width=500, center=true) }}
+{{ easy_image(src="joystick-protocols", width=500, center=true) }}
+{{ easy_image(src="joystick-axis-mapping", width=500, center=true) }}
+
 
 ##### MAVLink `MANUAL_CONTROL` Messages
 
 [`MANUAL_CONTROL`](https://mavlink.io/en/messages/common.html#MANUAL_CONTROL) MAVLink messages are 
-automatically sent to the vehicle at 25Hz, which is not currently configurable.
+automatically transmitted at 25Hz, which is not currently configurable.
+
+{% note() %}
+While it is generally expected for there to be a vehicle connected, Cockpit is capable of sending
+MAVLink Messages to a MAVLink system more generally, even without a vehicle included.
+{% end %}
+
+{% warning() %}
+If the browser tab is changed away from Cockpit, or the Cockpit application window is minimised,
+Cockpit loses access to the joystick inputs, and normally stops sending `MANUAL_CONTROL` messages
+to reflect the loss of input. This can trigger an autopilot failsafe, which may not be desirable,
+so an option is provided in the top right of the joystick page to tell Cockpit to repeatedly send
+the last joystick input it received before losing access to the joystick. 
+>
```

Comment:
It would be interesting to mention that one can avoid this kind of problem by having only the Cockpit tab in a given window and opening other tabs in another browser window.

It can also be avoided by using the self-contained version and not minimizing it (can be out of focus without a problem).

Reply:
Those are somewhat implied as-is, and seem out of scope of that warning to include specific details of.